File: 0056-merge-intervals/0056-merge-intervals.java
class Solution {
    public int[][] merge(int[][] intervals) {

        List<int[]> res = new ArrayList<>();
        Arrays.sort(intervals, (a,b) -> a[0] - b[0]);

        res.add(intervals[0]);
        int n = intervals.length;
        for(int i=1;i<n;i++){
            int[] prev = res.get(res.size() -1);
            int[] current = intervals[i];
            if(current[0] <= prev[1]){
                prev[1] = Math.max(prev[1], current[1]);
            }else{
                res.add(current);
            }
        }

        return res.toArray(new int[res.size()][]);
        
    }
}

File: 0056-merge-intervals/0056-merge-intervals.js
/**
 * @param {number[][]} intervals
 * @return {number[][]}
 */
var merge = function(intervals) {

    intervals.sort((a,b) => a[0] - b[0]);

    let res = [intervals[0]];

    for(let i=0;i<intervals.length;i++){
        let last = res[res.length -1];
        let current = intervals[i];

        if(current[0] <= last[1]){
            last[1] = Math.max(last[1], current[1]);
        }else{
            res.push(current);
        }
    }

    return res;
    
};